File: Java/ContainerWithMostWater.java
//
//  LeetCode
//  Created by Abdelrahman Nasser
//

class Solution {
    public int maxArea(int[] height) {
        int max = 0;
        int left = 0;
        int right = height.length - 1;

        while (left < right) {
            if (height[left] > height[right]) {
                int currentValue = height[right] * ( right - left );
                max = (currentValue > max) ? currentValue : max;
                right -= 1;
            }else {
                int currentValue = height[left] * ( right - left );
                max = (currentValue > max) ? currentValue : max;
                left += 1;
            }
        }

        return max;
    }
}

File: Swift/ContainerWithMostWater.swift
//
//  LeetCode
//  Created by Abdelrahman Nasser
//

class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var max = 0
        var left = 0
        var right = height.count - 1

        while (left < right) {
            if height[left] > height[right] {
                let currentValue = height[right] * ( right - left )
                max = (currentValue > max) ? currentValue : max
                right -= 1
            }else {
                let currentValue = height[left] * ( right - left )
                max = (currentValue > max) ? currentValue : max
                left += 1
            }
        }

        return max
    }
}
